File: Data Structures & Algorithms/is-subsequence/submission-0.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int j=0,i=0;
        while(i<s.length() && j<t.length()){
            if(s[i] == t[j]){
                i++;
            }
            j++;
        }

        return i==s.length();
    }
};

File: Data Structures & Algorithms/reverse-integer/submission-0.py
class Solution:
    def reverse(self, x: int) -> int:
        revNo = 0
        sign = -1 if x < 0 else 1
        x = abs(x)

        while x > 0:
            digit = x % 10
            revNo = revNo * 10 + digit
            x = x // 10  # use integer division instead of float division

        revNo *= sign

        # 32-bit signed integer boundary check
        if revNo < -(2**31) or revNo > (2**31 - 1):
            return 0

        return revNo